pkg/porter/lifecycle.go

```diff
@@ -298,6 +305,31 @@ func (p *Porter) BuildActionArgs(ctx context.Context, installation storage.Insta
 	return args, nil
 }
 
+// ensureVPrefix adds a "v" prefix to the version tag if it's not already there.
+// Version tag should always be prefixed with a "v", see https://github.com/getporter/porter/issues/2886.
+// This is safe because "porter publish" adds a "v", see
+// https://github.com/getporter/porter/blob/17bd7816ef6bde856793f6122e32274aa9d01d1b/pkg/storage/installation.go#L350
+func ensureVPrefix(opts *BundleReferenceOptions) error {
+	if opts._ref == nil {
+		ref, err := cnab.ParseOCIReference(opts.Reference)
+		if err != nil {
+			return err
+		}
+		opts._ref = &ref
+	}
+	if strings.HasPrefix(opts._ref.Tag(), "v") {
+		// don't do anything if "v" is already there
+		return nil
+	}
+	ref, err := opts._ref.WithTag("v" + opts._ref.Tag())
+	if err != nil {
+		return err
```

Comment:
It would be amazing to add some text with these errors to provide additional context

Reply:
yep